StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement.swift

```diff
@@ -133,54 +133,26 @@ public final class EmbeddedPaymentElement {
             } catch {
                 return UpdateResult.failed(error: error)
             }
-            guard !Task.isCancelled else {
+            guard let self, !Task.isCancelled else {
                 return UpdateResult.canceled
             }
 
-            // Store the old payment option before we update self.formViewController
-            let oldPaymentOption = self?.paymentOption
-
-            // 2.1. Re-initialize embedded form view controller to update the UI to match the newly loaded data.
-            if let formPaymentMethodType = self?.selectedFormViewController?.selectedPaymentOption?.paymentMethodType {
-                self?.selectedFormViewController = EmbeddedFormViewController(configuration: configuration,
-                                                                intent: loadResult.intent,
-                                                                elementsSession: loadResult.elementsSession,
-                                                                shouldUseNewCardNewCardHeader: loadResult.savedPaymentMethods.first?.type == .card,
-                                                                paymentMethodType: formPaymentMethodType,
-                                                                previousPaymentOption: self?.selectedFormViewController?.selectedPaymentOption,
-                                                                analyticsHelper: analyticsHelper)
-
-            }
-
```

Comment:
Bug was here - we didn't set delegate!  I made it a required param on the initializer so we can't forget it.